### ISSUE TITLE
`cfg` should warn if condition is meaningless

### ISSUE THREAD

**Issue by aturon** [rust-lang/rust#48885]
I accidentally wrote:

```rust
#[cfg(features = "unstable-futures")]
```

rather than `feature`, but had no heads-up that the `cfg` was meaningless. Took a while to track down. Can we warn in this case?

**Comment by sfackler** [comment]
You can add arbitrary cfgs through build scripts, so we can't really say that any cfg is meaningless.

**Comment by clarcharr** [comment]
Perhaps usage of `features` could warn to instead use `feature`?

**Comment by rkruppe** [comment]
Someone could pass `--cfg features=blah` though. Even warning about cfgs where the key is not set at all (with any value) is problematic because, e.g., `cfg(feature=..)` would warn if Cargo features exist but none are enabled in one compilation.

**Comment by euclio** [comment]
This might make sense as a clippy lint where you could provide a whitelist.

**Comment by Wenzel** [comment]
I also made the same mistake tonight, and lost 30 min tracking down the issue.
If rustc cannot print a warning about this, maybe clippy is the solution then ?

**Comment by Mark-Simulacrum** [comment]
I agree that in practice we can't do this in rustc since it is too false-positive prone so I'm going to close this; if someone wants to pursue this I would either go through the RFC process to try and argue the case for a false-positive prone lint here or add it to clippy.
